contracts/BaseToken.sol

```diff
@@ -32,7 +32,7 @@ contract BaseToken is IBaseToken, IIndexPrice, VirtualToken, BlockContext, BaseT
     ) external initializer {
         __VirtualToken_init(nameArg, symbolArg);
 
-        uint8 priceFeedDecimals = IPriceFeedV2(priceFeedArg).decimals();
+        uint8 priceFeedDecimals = IPriceFeedDispatcher(priceFeedArg).decimals();
```

Comment:
> 1. When deploying or upgrading a baseToken, it will have one PriceFeedDispatcher contract, one ChainlinkPriceFeedV3 contract, and one UniswapV3PriceFeed contract?

Yes

> 4. When pausing a market, we don't use EmergencyPriceFeed anymore right?

Yes

@twxia @tienshaoku  could you help to answer Q2 and Q3?

Reply:
@vinta 2 yes i'll discuss it with William!
3 thx for pointing that out! will do!

Reply:
for 3., 

I have replied in https://github.com/perpetual-protocol/perp-curie-periphery-contract/pull/35/files#r1053484472.

TLDR, limit order needs historical data which is not supported in v3. probabbly remain the same is fine.

Reply:
@tienshaoku @twxia Please check the reply https://github.com/perpetual-protocol/perp-curie-periphery-contract/pull/35/files#r1053936603 when you have time.